src/components/product-sorting-select/product-sorting-select.tsx

```diff
@@ -0,0 +1,58 @@
+import { useCallback, useMemo } from 'react';
+import { useSearchParams } from '@remix-run/react';
+import { getProductSortByFromUrlSearchParams, SORT_BY_SEARCH_PARAM } from '~/api/product-sorting';
+import { ProductSortBy } from '~/api/types';
+import { useOptimisticSearchParamsState } from '~/utils/use-optimistic-search-params-state';
+import { Select, SelectItem } from '../select/select';
+
+import styles from './product-sorting-select.module.scss';
+
+const sortingOptions: { value: ProductSortBy; label: string }[] = [
+    { value: ProductSortBy.newest, label: 'Newest' },
+    { value: ProductSortBy.priceAsc, label: 'Price (low to high)' },
+    { value: ProductSortBy.priceDesc, label: 'Price (high to low)' },
+    { value: ProductSortBy.nameAsc, label: 'Name A-Z' },
+    { value: ProductSortBy.nameDesc, label: 'Name Z-A' },
+];
+
+export const ProductSortingSelect = () => {
+    const [searchParams, setSearchParams] = useSearchParams();
+
+    const appliedSortBy = useMemo(
```

Comment:
Would it be a good idea to move sorting to separate hook to make it possible to use in other projects?

Reply:
but it is already a component. Ok.